stjude/proteinpaint
# Cross-origin session recovery in a new tab

## Description
Improvements: 
- There is now a menu option to delete cached sessions.
- The open UI now has a table of session names/IDs, the table renderer is also used in the delete option
- Clicking on a newly created `Share` URL link will open a new tab with the embedder's URL, no need to open the proteinpaint host site as an intermediary browser tab 

Test Scenarios for `Open` and `Save`: for each scenario, test that you can save to/open from browser, file, and if applicable, server.

A. Unrpotected ds: comment `serverconfig.dsCredentials.SJLife` (e.g., change to `dsCredentials['#SJLife']`):
  - use [a simple SJLife URL params](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg38%22,%22dslabel%22:%22SJLife%22})
  - there should be no server options in the `Open` and `Save` menus 

B. Protected ds: enable `dsCredentials.SJLife`
  - use http://localhost:3000/example.auth.html
  - there should **be** server options in the `Open` and `Save` menus
  - test while logged in and not
  - when saving to server, you should see the file saved in `[cachedir]/sessionsByCred/...`

C. Same as B, but simulate an embedded use case
  - `sudo vim /etc/hosts` and add the entry `127.0.0.1 viz.localhost`
  - change `dsCredentials['SJLife']` to `dsCredentials['*']` - all embedders will use the same jwt secret
  - from the `sjpp` dir, run `./proteinpaint/server/test/fake-jwt.js SJLife > public/fake-jwt.json '*'`
  - use http://viz.localhost:3000/example.auth.html (different subdomain)
  - manually test like in B
  - Also test that the `Share`-generated URL opens within `viz.localhost`, and not in `localhost`


## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
